hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/FieldReaderAppender.java

```diff
@@ -45,6 +45,8 @@ public Size apply(
 			MethodVisitor methodVisitor,
 			Implementation.Context implementationContext,
 			MethodDescription instrumentedMethod) {
+		FieldDescription.InDefinedShape persistentFieldAsDefined = persistentField.asDefined();
```

Comment:
You could move this to the constructor and store this as a attribute of  `FieldReaderAppender`, alongside `persistentField`. I think you would avoid a bit of noise if you did, in particular in the `fieldRead`/`fieldWrite` methods.